### ISSUE TITLE
User can write to system partition after uncheck him/her from share->priveleges

### ISSUE THREAD

**Issue by 3ed** [OpenMediaVault-Plugin-Developers/openmediavault-sftp#8]
# Step to reproduce:

- go to share->privileges
- remove existed checkbox or check it to 'no access'
- apply configuration

### Whats going to happen?

- Before:
/sftp/joedoe/sharename/[files from mounted share]
- After:
/sftp/joedoe/sharename/[empty rw on system partition]

If this folder have 755 then user can write to system partition!

# Proposed fix.

/usr/share/openmediavault/mkconf/fstab.d/91-sftp
```
--- a/91-sftp	2017-09-12 19:46:08.588903722 +0200
+++ b/91-sftp	2017-09-12 19:46:53.556032558 +0200
@@ -57,6 +57,7 @@
 
     if [ -n "${sfpath}" ] && [ -n "${username}" ] && [ -n "${option}" ]; then
         mkdir -p "/sftp/${username}/${sfname}"
+        chmod a=x "/sftp/${username}/${sfname}"
         echo "${sfpath2} /sftp/${username}/${sfname2} none bind,${option} 0 0"
     fi
 done
```
Set `${sfname2}` folder to chmod 111.

But ideally this folder should be deleted.

**Comment by ryecoaaron** [comment]
The plugin can trigger on a privilege change but I don't want to delete the folder since the share still exists in the sftp plugin.  Changing permissions is not ideal but I will have to think about this.

**Comment by 3ed** [comment]
bind will replace dstdir permissions with this from srcdir. Anyway, tree inside chroot should be as immutable as possible, except mounted (in term of whitelist policy).

**Comment by ryecoaaron** [comment]
Fixed in 3.1.4.  If you remove the privileges, it doesn't delete the sftp share but it does remove the bind mount and directory.  If you add the privileges back, it will create the bind mount and directory again.
